src/transformers/models/falcon/modeling_falcon.py

```diff
@@ -540,6 +766,7 @@ def forward(
         head_mask: Optional[torch.Tensor] = None,
         use_cache: bool = False,
         output_attentions: bool = False,
+        padding_mask: Optional[torch.LongTensor] = None,
```

Comment:
for reviewers: this is needed https://github.com/huggingface/transformers/pull/25598/commits/db8bd6407c122e59c285571c774aa8e14f6143fb as previously I used an approach where we create back the padding mask from the attention causal mask: https://github.com/younesbelkada/transformers/blob/0e30d13c903cd0b44545b59e6c7b8bfdd592f862/src/transformers/models/llama/modeling_llama.py#L72 

The `_convert_to_padding_mask` creates a huge overhead in the computation for padded input, making this PR unusable even for inference

Before 

![image](https://github.com/huggingface/transformers/assets/49240599/f92596fb-2738-4485-8036-20d3b01c7ab3)

with the padding_mask technique: 

![image](https://github.com/huggingface/transformers/assets/49240599/b34a5ff0-2452-4412-9de5-70532f4d7ce2)

Reply:
ok for me